### DISCUSSION TITLE
PCB

### DISCUSSION THREAD

**Discussion by geraldFS** [philibertc/micronova_controller#17] [Ideas]
Hi there,

I took the liberty of creating a new board based on your schematic.

What is different:
a) WEMO D1 Mini is supplied with 5V
either with external feed, or if two jumpers are closed with the 5V from the Mirconova

b) the 5V voltage is suppressed / smoothed by means of capacitors
C1 = 220µF
C2 = ~ 100nF
C3 = ~ 10nF
C4 = ~ 10þF

I have always noticed that when the voltage is not clean, the ESP now and then produces strange errors.

c) In addition, the 3.3V voltage can also (if necessary) be eliminated / smoothed
C5 = ~ 100nf
C6 = ~ 10þF

d) The 18V are brought out separately if an external DC / DC module is to be used.

e) Pin D6 is brought out to connect a 1Wire temperature sensor (e.g. room temperature remote) for free use

f) Pins D8 and D7 are prepared for further use (i2C bus), as are taps in front of 3.3V / GND

The software implementation of the 1Wire connection would have to be added.

The hardware layout has not yet been tested, I will first have a few manufactured and tested.
There are probably brave ones who check it and test it if necessary.

Regards

Gerald

![Pellet_1_Leiterplatte](https://user-images.githubusercontent.com/35431341/137638746-35aea1ef-f854-4136-b317-5220b399bb82.png)

VCC 5 = GND/VCC5/VCC5 screw (terminal therefore 3 pin)
VCC = Jumper close 5V from Mirconova / open externe VCC
GND = Jumper close GND (Mirconova)
GND | 18V = GND / 18V from Mirconova (screw terminal therefore 3 pin)
SERIAL = From Mirconova (18V | 5V | GND | Stove Serial)
RESET = Jumper Close (Reset the Firmware)
1Wire = GND | Data (D6) | 3,3V

**Comment by philibertc** [comment]
Good evening,
Thank you very much for these contributions!

Just a clarification for the 5V: it is a power line for the "small" electronics of the stove, and I don't know how much it can provide, so I preferred to use the 20V line which is the general power supply.
(I have an official MicroNova WiFi module and although its electronics are in 5V it is powered via a buck converter on the 20V).

For all the additional implementations, I will add them in my version 2 of the PCB.

**Reply by geraldFS** [reply]
I would like to provide you with the layout (Fritzing).

**Reply by philibertc** [reply]
I use KiCAD but don't worry, I could do this quickly.

**Reply by geraldFS** [reply]
Exactly for this reason I executed the supply twice (external 5V either power supply / or via DC / DC from the 18V) plus the possibility to use the internal 5V.

**Reply by philibertc** [reply]
Hello, here is the V2.
I took some of your ideas (like the screw connector, the possibility to do a reset and to add other components), but I modified and added some things:
- there is a connection between D0 and RST to allow deep sleep (very soon implemented in the firmware)
- the reset can be done with a flat screwdriver easily
- for the addition of components, I preferred not to impose a ready-made wiring for 1Wire, I²C: I leave more freedom here
- for all this smoothing with capacitors, there is no need because people use either the 3.3V input (through a buck converter from 20V) or the USB port. And whether it's in a USB power supply or a buck converter there is the necessary smoothing

![image](https://user-images.githubusercontent.com/57588282/138329627-8cf75ac9-88bd-4fa1-9a52-bbca8f955eab.png)
![image](https://user-images.githubusercontent.com/57588282/138329791-f4ece532-98fc-457a-a9f1-5abca85b4ccf.png)

**Comment by twostroker84** [comment]
Hello
I am anxiously waiting for the pcb to be available for sale again.
If you like, contact me as soon as it is available!
Beautiful project
Bye!

**Reply by philibertc** [reply]
Hello ! @twostroker84
I have started the production of a series of PCBs, I should have them in two weeks
Sincerely,
philibertc

**Comment by testeur70** [comment]
Hello Philibert,

Thanks you for this great job. I am trying to build your v1 pcb and I wonder where you connect the "IN+" and "IN-" cables from the LM2596. The "IN+" should go to  20 v on the stove ? But the"IN-" can be connected to the GND on the Pcb? I can not see exactly those connections on your tindie pictures.
Thanks for your answer.

**Reply by philibertc** [reply]
Hello,
IN- must be connected to the GND of the stove, which can be retrieved from the GND at the bottom right of the PCB.
![image](https://user-images.githubusercontent.com/57588282/139286951-2be2ae4c-a61f-4fc9-8c09-dab3ff95e5c0.png)
IN+ should be connected to the 20V of the stove, preferably via a resettable fuse.

**Reply by testeur70** [reply]
Merci
;-)

**Comment by phoffi** [comment]
With the Buck Converter this could also be a feature for next PCB version?

Soemthing like that mini buck converter, directly soldered on PCB:

https://aliexpress.com/item/32998937888.html

**Reply by philibertc** [reply]
Hello, yes it could be practical. But these converters don't have a very clean output, and some people prefer to use USB.

**Comment by testeur70** [comment]
Hello Philibertc,

Quel sont les 3 composants entourés sur le pcb de la v2 ( voir photo) ce
sont des jumper ?
Merci pour ton retour

**Reply by philibertc** [reply]
Hello, are you talking about the three components marked PC817?
![image](https://user-images.githubusercontent.com/57588282/145706804-132dd43a-8af9-4668-b5af-a213924bea18.png)
I am speaking in English so that everyone can follow the thread.

**Comment by testeur70** [comment]
Hello Philiberc, Sorry for asking this question in french, as I havent
practice my english for a long time. I know that PC817 are optocoupler. My
question is about the modifications you made between V1 and V2. On the
picture joined, it is the three red parts I noticed on the PCB. I don't
know what thery are ? Jumper or nothing ?
Thanks for your answer

**Reply by philibertc** [reply]
I don't see what image you are talking about, did you forget to attach the picture?
Je ne vois pas de quelle image vous parlez, vous avez oublié de joindre l'image ?
PS: Vous pouvez poser les questions en français, je pourrais rajouter la traduction en éditant vos messages.

**Comment by testeur70** [comment]
Hello I don't understand why my picture can not be sent.  Maybe the size is
too big. You can find it following this ink :

https://i25.servimg.com/u/f25/19/33/25/06/pcb10.png
Thanks for your help

**Reply by philibertc** [reply]
![image](https://user-images.githubusercontent.com/57588282/145712625-17e81c36-a812-4052-a019-b43f086a870e.png)
This one is used to reset the parameters of the microcontroller by putting a flat screwdriver on it, it just replaces a push button.
![image](https://user-images.githubusercontent.com/57588282/145712661-6a21b9f6-c247-44b2-8bc2-b8bf1a50e15d.png)
These two are used to connect male or female pin strips, to connect cables.
![image](https://user-images.githubusercontent.com/57588282/145712716-8481da08-f687-4283-b014-a76b777c347f.png)
The upper one is connected to the upper terminal footprint and the lower one to the lower connector, so you can choose between using a pin strip or screw terminals.

**Reply by testeur70** [reply]
Thanks for your answer. The explanations are very clear now. Do you have a pictures with all components soldered and Interface connected ?

**Reply by philibertc** [reply]
I unfortunately only have pictures of version 1:
![image](https://user-images.githubusercontent.com/57588282/145722847-0618aa0e-bcbb-4ddf-85b4-b641acd6e374.png)
but you can imagine the new layout with this scheme:
![image](https://user-images.githubusercontent.com/57588282/145722911-4bee168b-4cbc-46da-ae9b-9a576b33426f.png)

**Comment by testeur70** [comment]
Thanks

**Comment by foxmt** [comment]
Hi Philibert

I've ordered you 2 pcb with kit.

![image](https://user-images.githubusercontent.com/11096252/217233685-72a404c9-497d-45de-a566-407f47cafa12.png)

 I don't understand where 3.3v must be soldered. Does i need a separated 3.3v power ? II thought the 3.3V was provided by the card

![image](https://user-images.githubusercontent.com/11096252/217234083-e80a972f-cb34-49fb-a6f3-e7295df37908.png)

**Reply by foxmt** [reply]
Is 3.3V (Power in), is only a pin test ?

**Reply by philibertc** [reply]
Hello,
the "Power in" inputs (3V3 and GND) are used to power the circuit without using the micro USB input of the ESP. On one of the 4 pins of the "Serial" port of the stove, there is one capable of powering an external circuit in 20V.
So to power the controller you can use a classic power supply or a LM2596 module to lower the 20V and power the controller through the "Power in" input.

Bests,
philibertc
